### PR TITLE
Checkout page

### PR DESCRIPTION
**Pull request by PuneetP16**
Features added: 
Checkout Page,
Toastify (For Toasts),
refactor CSS, dark theme and responsiveness.
Address Management ( Add and Delete feature)
On clicking the checkout button, the dummy order gets placed and the cart is off-loaded. 

image

image

image

On selection of the address the checkout button becomes visible:
image

On removal of the address the checkout button becomes invisible:
image

in dark mode:
image

### CODE REVIEW

File: src/components/Card/AddressCard/AddressCard.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useAuth } from "../../../contexts";
+import { useAxios } from "../../../customHooks";
+import { bxIcons } from "../../../data/icons";
```
Comment (saurabhchirde) [inline src/components/Card/AddressCard/AddressCard.jsx:3]: try to use absolute path

File: src/components/UI/Modal/Modal.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.modal__wrapper {
+	position: fixed;
+	top: 50%;
+	left: 50%;
+	width: 100%;
+	height: 100%;
+	transform: translate(-50%, -50%);
+	backdrop-filter: blur(2px);
+	z-index: var(--z-index-3);
+	background-color: rgba(0, 0, 0, 0.45);
```
Comment (saurabhchirde) [inline src/components/UI/Modal/Modal.css:10]: use variables

File: src/components/Card/OverviewCard/OverviewCard.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { Rating } from "../../../components";
+import "./OverviewCard.css";
+import { useAxios } from "../../../customHooks";
+import { useCart, useWishlist } from "../../../contexts";
+import { useState } from "react";
```
Comment (saurabhchirde) [inline src/components/Card/OverviewCard/OverviewCard.jsx:5]: absolute path

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 4]
```diff
@@ -6,7 +6,8 @@ import {
 	Alert,
 	InputTypeTwo,
 } from "../../../components";
-import { useUser } from "../../../contexts";
+import { Toast } from "../../../components/UI/Toast/Toast";
+import { useTheme, useUser } from "../../../contexts";
 import { useDocumentTitle } from "../../../customHooks";
 import { signUp } from "../../../services";
```
Comment (saurabhchirde) [inline src/pages/Authentication/SignUp/SignUp.jsx:12]: here as well

**Review by saurabhchirde — APPROVED** [review]
Gave some minor suggestions. Otherwise, your code didn't have any issues. You can proceed to merge.
As per users' order flow, when the user is placing an order, you can add a confirmation popup with the order number or some short order summary to let the user know that the order has been placed. Also, provide a way to check their orders by providing a link to the all orders section.

File: src/components/Card/OverviewCard/OverviewCard.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+export const OverviewCard = ({ prod, btnTitle, featured }) => {
+	const [disable, setDisable] = useState(false);
+	const { axiosRequest } = useAxios();
+	const { removeFromCart, updateQtyInCart } = useCart();
+	const { addToWishlist, wishlist } = useWishlist();
+
+	const isProductInWishlist =
+		wishlist.findIndex((p) => p._id === prod._id) !== -1;
```
Comment (m26unkwn) [inline src/components/Card/OverviewCard/OverviewCard.jsx:14]: you can take this function out of the context

File: src/components/UI/Modal/Modal.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		if (e.target.classList.contains("modal__wrapper")) {
+			setModal(false);
+		}
+	};
+	return (
+		<div
+			className={`${
+				modalClass ? modalClass : "modal__wrapper"
+			} modal__wrapper_center_child`}
+			// onClick={hideModal} kept for development purpose
```
Comment (m26unkwn) [inline src/components/UI/Modal/Modal.jsx:16]: now you can opt out before pushing this code.

File: src/contexts/authContext.js [hunk 7]
```diff
@@ -1,18 +1,79 @@
...
 
 export const useAuth = () => useContext(AuthContext);
 
+const initialAddressState = {
+	addresses: [],
+};
+
 export const AuthProvider = ({ children }) => {
 	const token = localStorage.getItem("storeToken");
+	const { theme } = useTheme();
```
Comment (m26unkwn) [inline src/contexts/authContext.js:16]: write this code outside of the provider

**Review by m26unkwn — APPROVED** [review]
Your code is readable and clean

you can look for the suggestion given and can merge the code.

File: src/components/UI/Modal/Modal.css (hunk 2 above)
Comment (PuneetP16) [reply src/components/UI/Modal/Modal.css:10]: Done it.

File: src/components/Card/OverviewCard/OverviewCard.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/components/Card/OverviewCard/OverviewCard.jsx:5]: Will look into it.

File: src/contexts/authContext.js (hunk 7 above)
Comment (PuneetP16) [reply src/contexts/authContext.js:16]: Ok.

**Pull request merged**